lib/LoginHookListener.php

```diff
@@ -93,23 +86,23 @@ private function matchCidr($ip, $range) {
 				case 3:
 					$binMask .= "e";
 					break;
-			 }
+			}
 
-			 $binMask = str_pad($binMask, 32, '0');
-			 $binMask = pack("H*", $binMask);
+			$binMask = str_pad($binMask, 32, '0');
+			$binMask = pack("H*", $binMask);
 
-			 if ( ($ip & $binMask) === $subnet ) {
-				 return true;
-			 }
+			if (($ip & $binMask) === $subnet) {
+				return true;
+			}
 		}
 		return false;
-	 }
+	}
 
 	/**
 	 * @return bool
 	 */
-	public function isLoginAllowed() {
-		$allowedRanges = $this->config->getAppValue('limit_login_to_ip', 'whitelisted.ranges', '');
+	public function isLoginAllowed(): bool {
+		$allowedRanges = (string) $this->config->getAppValue('limit_login_to_ip', 'whitelisted.ranges', '');
```

Comment:
It always returns string

Reply:
`psalm` agrees with you ^^
I needed it for tests (bad mock I suppose), but I plan to fix it :)